scripts/concatenate_files.py

```diff
@@ -0,0 +1,69 @@
+import json
+import os
+import re
+import yaml
+
+docs_root = "./docs"
+
+
+def concat(of, level, doc_path):
+    full_path = f"{docs_root}/{doc_path}.md"
+    if not os.path.exists(full_path):
+        full_path = f"{docs_root}/{doc_path}/index.md"
+
+    with open(full_path, "r") as doc_file:
+        doc_text = doc_file.read()
+        # parse YAML header
+        yaml_match = re.match("^---$((.|\n)*?)^---$", doc_text, flags=re.MULTILINE).groups(1)
```

Comment:
I think regex is overcomplicating this, just do a split with a limit on `---\n`

Reply:
Good point, I simplified this